messaginginapp/src/main/java/io/customer/messaginginapp/ModuleMessagingInApp.kt

```diff
@@ -12,82 +14,40 @@ import io.customer.sdk.events.Metric
 
 class ModuleMessagingInApp(
     config: MessagingInAppModuleConfig
-) : CustomerIOModule<MessagingInAppModuleConfig> {
+) : CustomerIOModule<MessagingInAppModuleConfig>, GistListener {
     override val moduleName: String = MODULE_NAME
     override val moduleConfig: MessagingInAppModuleConfig = config
 
     private val diGraph: SDKComponent = SDKComponent
     private val eventBus = diGraph.eventBus
-    private val gistProvider by lazy { diGraph.gistProvider }
+    private val gistProvider: GistProvider
+        get() = diGraph.gistProvider
```

Comment:
I see a lot similar usages for `get` in the PR, is there a reason for preferring `get()` over following:

```
private val gistProvider: GistProvider = diGraph.gistProvider
```

Particularly when `gistProvider` is singleton and is used so many times in the class, I think using `get()` is slightly inefficient here 🤓

Reply:
The benefit of having this is, on compile-time behavior, this declaration does not set any value
It simply defines a property with a custom-getter and usage of it is very limited so even if recomputes, the value is `Singleton` so it brings back the old value without creating a new instance. 

`GistProvider` takes in an `application` instance which is dependent on the SDK lifecycle, so want to only access its value when it's about to get used instead of accessing it as compile time. 

We can use `lazy` here too, but it does the exactly same thing,
- do not initialize at compile time
- stores the value in a parameter and then checks if it exists and returns it.

Reply:
Discussed on call, we'll continue using `get` for classes accessing `AndroidSDKComponent` to keep them lazy and updated for others.